lib/RoleDetails/RoleDetails.js

```diff
@@ -106,21 +106,23 @@ export const RoleDetails = ({
       || !isTargetTenantCentral
       || isRoleShared
     );
-    const isActionMenuHidden = isMutationsPrevented && isSharingPrevented;
+
+    const isEditAllowed = isRoleShared ? stripes.hasPerm('consortia.sharing-roles-all.item.post') : stripes.hasPerm('roles.item.put');
+    const isCreateAllowed = stripes.hasPerm('roles.item.post');
+    const isDeleteAllowed = isRoleShared ? stripes.hasPerm('consortia.sharing-roles-all.item.delete') : stripes.hasPerm('roles.item.delete');
+    const isActionMenuHidden = (isMutationsPrevented && isSharingPrevented) || (!isEditAllowed && !isCreateAllowed && !isDeleteAllowed);
```

Comment:
This is a nice setup that makes the otherwise twisty logic of ECS or single-tenant, etc etc, difficult to follow.